### PR TITLE
add stateful single-node KES implementation

### PR DESCRIPTION
**Pull request by aead**
This commit adds support for stateful KES deployments.

The main sections of this commit are:
 - Stateful KES initialization - i.e. `kes init` commands
 - FS Vault implementation that supports multiple enclaves
 - FS encoding of data using the generalized protobuf encoding
   in the standard lib (encoding/gob).
 - FS encryption using `Wrap` / `Unwrap` functions.

This commit only adds support for single-node stateful KES
deployments. Support for distributed clusters of multiple
KES nodes will be built on top of this standalone implementation.

### CODE REVIEW

File: cmd/kes/server.go [hunk 1]
```diff
@@ -347,6 +376,137 @@ func serverCmd(args []string) {
...
+	quiet := quiet(serverConfig.Quiet)
+	quiet.Print(blue.Sprint("Endpoint: "))
+	quiet.Println(bold.Sprint(alignEndpoints(margin, listeningOnV4(ip), port)))
+	quiet.Println()
+
+	// if r, err := hex.DecodeString(config.Admin.Identity.Value().String()); err == nil && len(r) == sha256.Size {
+	// 	quiet.Println(blue.Sprint("Admin:   "), config.Admin.Identity.Value())
+	// } else {
+	//	quiet.Println(blue.Sprint("Admin:   "), "_     [ disabled ]")
+	// }
```
Comment (Alevsk) [inline cmd/kes/server.go:482]: Can this commented code be removed?

File: cmd/kes/server.go [hunk 2]
```diff
@@ -347,6 +376,137 @@ func serverCmd(args []string) {
...
+	if init.VerifyClientCerts.Value() {
+		quiet.Println(blue.Sprint("Auth:    "), color.New(color.Bold, color.FgGreen).Sprint("on "), color.GreenString("  [ only clients with trusted certificates can connect ]"))
+	} else {
+		quiet.Println(blue.Sprint("Auth:    "), color.New(color.Bold, color.FgYellow).Sprint("off"), color.YellowString("  [ any client can connect but policies still apply ]"))
+	}
+	quiet.Println()
+
+	// quiet.Println(blue.Sprint("Keys:    "), fmt.Sprintf("%s: %s", kmsKind, kmsEndpoint))
```
Comment (Alevsk) [inline cmd/kes/server.go:490]: Can this commented code be removed?

**Review by Alevsk — APPROVED** [review]

File: cmd/kes/server.go (hunk 1 above)
Comment (Alevsk) [inline cmd/kes/server.go:482]: Can this commented code be removed?

File: cmd/kes/server.go (hunk 2 above)
Comment (Alevsk) [inline cmd/kes/server.go:490]: Can this commented code be removed?

**Review by Alevsk — APPROVED** [review]

File: cmd/kes/server.go (hunk 1 above)
Comment (aead) [reply cmd/kes/server.go]: 👍

**Review by Alevsk — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**